### PR TITLE
Changed =='\0' to .isEmpty() in multiple lines Resolves issue #145

### PR DESCRIPTION
**Pull request by techmagellc**
Change is needed to allow compile under gcc >=3.0

### CODE REVIEW

**Comment by juampe** [comment]
To compile again please use   @2.6.3 pio espressif8266 framework

```
[env:WEMOS_D1_Mini]
platform = espressif8266 @2.6.3
board = d1_mini
framework = arduino
lib_deps = ${common.lib_deps_ext}
monitor_speed = 115200
upload_speed = 460800
```

**Review by toaday — APPROVED** [review]

**Comment by scottmmjackson** [comment]
@juampe that workaround does not work.

The PR fixes the issue. Is there a reason it can't be merged?

**Comment by juampe** [comment]
@scottmmjackson  sorry but I don't have permissions to commit in this repository, I only provide a "forzen in time" workaround. That last activity in this repo was in 24 Feb. I have submited PR in the past and was accepted. @gysmo38 Are you OK?
PD I think that it will an interesting thing continue the development process in another fork. I'm already interested to test some new functionalities.

File: src/mitsubishi2mqtt/mitsubishi2mqtt.ino [hunk 1]
```diff
@@ -269,22 +269,22 @@ void saveUnit(String tempUnit, String supportMode, String loginPassword, String
...
-  if (maxTemp == '\0') maxTemp = 31;
+  if (maxTemp.isEmpty()) maxTemp = 31;
   doc["max_temp"]   = maxTemp;
   // if tempStep is empty, we use default 1
-  if (tempStep == '\0') tempStep = 1;
+  if (tempStep.isEmpty()) tempStep = 1;
   doc["temp_step"] = tempStep;
   // if support mode is empty, we use default all mode
-  if (supportMode == '\0') supportMode = "all";
+  if (supportMod.isEmpty()) supportMode = "all";
```
Comment (plastbox) [inline src/mitsubishi2mqtt/mitsubishi2mqtt.ino:284]: Spelling error. `supportMod` !== `supportMode`

**Comment by gysmo38** [comment]
Thank you :)

**Pull request merged**